pyscf/adc/radc.py

```diff
@@ -1254,258 +1187,226 @@ def get_imds_ip(adc, eris=None):
-    M_ij += 0.25 *  lib.einsum('i,lide,jlde->ij',e_occ,t2_1, t2_1, optimize=True)
-    M_ij -= 0.25 *  lib.einsum('i,lide,ljde->ij',e_occ,t2_1, t2_1, optimize=True)
-    M_ij -= 0.25 *  lib.einsum('i,ilde,jlde->ij',e_occ,t2_1, t2_1, optimize=True)
-    M_ij -= 0.25 *  lib.einsum('i,ilde,jlde->ij',e_occ,t2_1, t2_1, optimize=True)
-
-    M_ij -= 0.25 *  lib.einsum('j,ilde,jlde->ij',e_occ,t2_1, t2_1, optimize=True)
-    M_ij += 0.25 *  lib.einsum('j,ilde,ljde->ij',e_occ,t2_1, t2_1, optimize=True)
-    M_ij += 0.25 *  lib.einsum('j,lide,jlde->ij',e_occ,t2_1, t2_1, optimize=True)
-    M_ij -= 0.25 *  lib.einsum('j,lide,ljde->ij',e_occ,t2_1, t2_1, optimize=True)
-    M_ij -= 0.25 *  lib.einsum('j,ilde,jlde->ij',e_occ,t2_1, t2_1, optimize=True)
-    M_ij -= 0.25 *  lib.einsum('j,ilde,jlde->ij',e_occ,t2_1, t2_1, optimize=True)
-
-    M_ij += 0.5 *  lib.einsum('ilde,jdle->ij',t2_1, eris_ovov,optimize=True)
-    M_ij -= 0.5 *  lib.einsum('lide,jdle->ij',t2_1, eris_ovov,optimize=True)
-    M_ij -= 0.5 *  lib.einsum('ilde,jeld->ij',t2_1, eris_ovov,optimize=True)
-    M_ij += 0.5 *  lib.einsum('lide,jeld->ij',t2_1, eris_ovov,optimize=True)
-    M_ij += lib.einsum('ilde,jdle->ij',t2_1, eris_ovov,optimize=True)
-
-    M_ij += 0.5 *  lib.einsum('jlde,idle->ij',t2_1, eris_ovov,optimize=True)
-    M_ij -= 0.5 *  lib.einsum('ljde,idle->ij',t2_1, eris_ovov,optimize=True)
-    M_ij -= 0.5 *  lib.einsum('jlde,ldie->ij',t2_1, eris_ovov,optimize=True)
-    M_ij += 0.5 *  lib.einsum('ljde,ldie->ij',t2_1, eris_ovov,optimize=True)
-    M_ij += lib.einsum('jlde,idle->ij',t2_1, eris_ovov,optimize=True)
+    M_ij_t = lib.einsum('ilde,jlde->ij', t2_1,t2_1, optimize=True)
+    M_ij -= lib.einsum('i,ij->ij',e_occ,M_ij_t, optimize=True)
+    M_ij -= lib.einsum('j,ij->ij',e_occ,M_ij_t, optimize=True)
+
+    M_ij_t = lib.einsum('ilde,ljde->ij', t2_1,t2_1, optimize=True)
+    M_ij += 0.5 * lib.einsum('i,ij->ij',e_occ,M_ij_t, optimize=True)
+    M_ij += 0.5 * lib.einsum('j,ij->ij',e_occ,M_ij_t, optimize=True)
+    del M_ij_t
 
+    M_ij += 0.5 *  lib.einsum('ilde,jdel->ij',t2_1, eris_ovvo,optimize=True)
+    M_ij -= 0.5 *  lib.einsum('lide,jdel->ij',t2_1, eris_ovvo,optimize=True)
+    M_ij -= 0.5 *  lib.einsum('ilde,jedl->ij',t2_1, eris_ovvo,optimize=True)
+    M_ij += 0.5 *  lib.einsum('lide,jedl->ij',t2_1, eris_ovvo,optimize=True)
+    M_ij += lib.einsum('ilde,jdel->ij',t2_1, eris_ovvo,optimize=True)
+
+    M_ij += 0.5 *  lib.einsum('jlde,idel->ij',t2_1, eris_ovvo,optimize=True)
+    M_ij -= 0.5 *  lib.einsum('ljde,idel->ij',t2_1, eris_ovvo,optimize=True)
+    M_ij -= 0.5 *  lib.einsum('jlde,ldei->ij',t2_1, eris_ovvo,optimize=True)
+    M_ij += 0.5 *  lib.einsum('ljde,ldei->ij',t2_1, eris_ovvo,optimize=True)
+    M_ij += lib.einsum('jlde,idel->ij',t2_1, eris_ovvo,optimize=True)
+
+    del t2_1
     cput0 = log.timer_debug1("Completed M_ij second-order terms ADC(2) calculation", *cput0)
     # Third-order terms
 
     if (method == "adc(3)"):
+      
```

Comment:
Please remove extra white spaces below and add white spaces where missing to separate blocks of code.

Reply:
Done